OvmfPkg/PlatformPei/PlatformId.c

```diff
@@ -0,0 +1,660 @@
+{
+  if ((Buffer > BufferEnd) || ((UINTN)(BufferEnd - Buffer) < Offset)) {
+    return EFI_BUFFER_TOO_SMALL;
+  }
+
+  *Result = Buffer + Offset;
+  return EFI_SUCCESS;
+}
+
+/**
+ * Reads a 32-bit integer from Buffer and returns it in *Value, and the pointer
+ * after the integer in Buffer in *End.
+ *
+ * @param[in] Buffer  The buffer to read from.
+ * @param[in] BufferEnd  The end of the buffer.
+ * @param[out] Value  The value read.
+ * @param[out] End  The byte after the value.
+ *
+ * @retval EFI_SUCCESS  The value was read successfully.
+ * @retval EFI_BUFFER_TOO_SMALL  The buffer is smaller than 32 bits.
+*/
+STATIC
+EFI_STATUS
+ParseUint32 (
+  IN  UINT8   *Buffer,
+  IN  UINT8   *BufferEnd,
+  OUT UINT32  *Value,
+  OUT UINT8   **End
+  )
+{
+  EFI_STATUS  Status;
+
+  Status = SafeOffset (Buffer, BufferEnd, sizeof (UINT32), End);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  *Value = *(UINT32 *)Buffer;
+  return EFI_SUCCESS;
+}
+
+/**
+ * Reads a first byte as a size, then confirms that Buffer + 1 + size
+ * does not exceed BufferEnd and that the string is NUL-terminated.
+ *
+ * @param[in] Buffer  The buffer to read from.
+ * @param[in] BufferEnd  The end of the buffer.
+ * @param[out] Start  The start of the string.
+ * @param[out] End  The byte after the NUL-terminator.
+ *
+ * @retval EFI_SUCCESS  The string was read successfully.
+ * @retval EFI_BUFFER_TOO_SMALL  The string's declared size was too large for
+ *                               the buffer.
+ * @retval EFI_INVALID_PARAMETER  The string was not NUL-terminated.
+ */
+STATIC
+EFI_STATUS
+ConsumeAsciiByteStr (
+  IN  UINT8  *Buffer,
+  IN  UINT8  *BufferEnd,
+  OUT UINT8  **Start,
+  OUT UINT8  **End
+  )
+{
+  EFI_STATUS  Status;
+  UINT8       EvtSize;
+
+  Status = SafeOffset (Buffer, BufferEnd, sizeof (EvtSize), Start);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  EvtSize = *Buffer;
+  // The spec does not allow for a zero-length string since it must be NUL-
+  // terminated.
+  if (EvtSize == 0) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "Expected a NUL-terminated string\n"));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  Status = SafeOffset (*Start, BufferEnd, EvtSize, End);
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  if (*(*End - 1) != 0) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "Expected a NUL-terminated string\n"));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+ * Reads a first 32-bit integer as a size, then determines if Buffer + 4 + size
+ * exceeds BufferEnd.
+ *
+ * @param[in] Buffer  The buffer to read from.
+ * @param[in] BufferEnd  The end of the buffer.
+ * @param[out] Start  The start of the string.
+ * @param[out] End  The byte after the last character of the string.
+ *
+ * @retval EFI_SUCCESS  The string was read successfully.
+ * @retval EFI_BUFFER_TOO_SMALL  The string's declared size was too large for
+ *                               the buffer.
+*/
+STATIC
+EFI_STATUS
+ConsumeUint32Str (

```

Comment:
The name `Uint32Str` is very confusing.
It could be interpreted as a wchar string with each wchar is 4 bytes.

Reply:
Removed for uninterpreted single file solution.